control-plane/gateways/deployment_init_container.go

```diff
@@ -30,15 +32,15 @@ type initContainerCommandData struct {
 	LogJSON  bool
 }
 
-// containerInit returns the init container spec for connect-init that polls for the service and the connect proxy service to be registered
+// initContainer returns the init container spec for connect-init that polls for the service and the connect proxy service to be registered
 // so that it can save the proxy service id to the shared volume and boostrap Envoy with the proxy-id.
-func initContainer(config GatewayConfig, name, namespace string) (corev1.Container, error) {
+func (b *meshGatewayBuilder) initContainer() (corev1.Container, error) {
```

Comment:
I either had to add a param to this function to pass in gateway class config or associate the function with the builder. I felt the association was cleaner because we are now using so much of the struct. 

It doesn't really need to be a pointer receiver. I think that might make it faster to not copy, but nothing is changed on the struct by the function.

Reply:
I think having consistency with the rest of the method set (where it's all defined as a pointer receiver) is another big reason to keep this as a pointer receiver

Reply:
Yes, I like the pointer reviever/pulling the params from the builder to keep everything in line with the rest of the package. 🚀

Reply:
the only downside is that this logic is then specifically associated with the `meshGatewayBuilder` when it will likely need to be consumed by `apiGatewayBuilder` and/or `terminatingGatewayBuilder` as well in the future. We can figure that out when the time comes though 👍